core/src/main/kotlin/fr/sncf/osrd/api/api_v2/pathfinding/PathfindingBlocksEndpointV2.kt

```diff
@@ -196,6 +197,19 @@ private fun computePaths(
     throw NoPathFoundException(NotFoundInBlocks(listOf(), Length(0.meters)))
 }
 
+fun findWaypointBlocks(
```

Comment:
It's weird that this method is added here and only used in stdcm. I haven't checked in details if that works out, but I feel like the lines 76-80 in this file can be replaced by a call to this method.

Reply:
Doesn't do the exact same thing. The method here returns them all into a set, the code you're referring to groups them by track.location. I'll just move this method to `STDCMEndpointV2` as a private method.